pkgs/applications/audio/bespokesynth/default.nix

```diff
@@ -50,70 +28,69 @@ let
-    runHook preConfigure
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=${buildType}"
+  ] ++ lib.optional enableVST2 "-DBESPOKE_VST2_SDK_LOCATION=${vst-sdk}/VST2_SDK";
 
-    export HOME=$(mktemp -d)
-    xvfb-run sh -e <<EOF
-      ${projucer}/bin/Projucer --set-global-search-path linux defaultJuceModulePath ${projucer.src}/modules
-      ${projucer}/bin/Projucer --resave BespokeSynth.jucer
-    EOF
-
-    runHook postConfigure
-  '';
-  CFLAGS = "-I${vst-sdk}/VST2_SDK";
-
-  nativeBuildInputs = [ xvfb-run pkg-config python3 makeWrapper ];
+  nativeBuildInputs = [ python3 makeWrapper cmake pkg-config ninja ];
 
+  # List obtained in https://github.com/BespokeSynth/BespokeSynth/blob/main/azure-pipelines.yml
   buildInputs = [
-    libX11 libXrandr libXinerama libXext libXcursor freetype libGL
-    ncurses libusb1
-    alsa-lib libjack2
+    libX11
+    libXrandr
+    libXinerama
+    libXext
+    libXcursor
+    curl
+    gtk3
+    webkitgtk
+    freetype
+    libGL
+    libusb1
+    alsa-lib
+    libjack2
+    gnome.zenity
+    alsa-tools
+    libxcb
+    xcbutil
+    libxkbcommon
+    xcbutilkeysyms
+    xcb-util-cursor
+    pcre
+    mount
+    patchelf
   ];
 
-  preBuild = ''
-    cd Builds/LinuxMakefile
-  '';
-  makeFlags = [ "CONFIG=${buildType}" ];
-  enableParallelBuilding = true;
-
-  installPhase = ''
-    runHook preInstall
-
-    mkdir -p $out/bin $out/share/bespokesynth $out/share/applications $out/share/icons/hicolor/512x512/apps
-    cp build/BespokeSynth $out/bin/
-    cp -ar ../MacOSX/build/Release/resource $out/share/bespokesynth/
-    wrapProgram $out/bin/BespokeSynth \
-      --run "cd $out/share/bespokesynth"
-
-    mkdir -p $out/share/applications/ $out/share/icons/hicolor/512x512/apps/
-    cp ../../bespoke_icon.png $out/share/icons/hicolor/512x512/apps/
-    substitute ../../BespokeSynth.desktop $out/share/applications/BespokseSynth.desktop \
-      --replace "/usr/bin/" ""
-
-    runHook postInstall
+  # Ensure zenity is available, or it won't be able to open new files.
+  # Ensure the python used for compilation is the same as the python used at run-time.
+  # jedi is also required for auto-completion.
+  postInstall = ''
+    wrapProgram $out/bin/BespokeSynth --prefix PATH : '${
+      lib.makeBinPath [
+        gnome.zenity
+        (python3.withPackages (ps: with ps; [ jedi ]))
+      ]
+    }'
   '';
 
   meta = with lib; {
-    description = "Software modular synth with controllers support, scripting and VST";
+    description =
+      "Software modular synth with controllers support, scripting and VST";
     homepage = "https://github.com/awwbees/BespokeSynth";
     license = with licenses; [
       gpl3Plus
-
-      # This package is unfree and not distributable due to the license of VST2.
-      # see #145607
-      unfree
-    ];
-    maintainers = with maintainers; [ astro ];
+    ] ++ lib.optional enableVST2 unfree;
+    maintainers = with maintainers; [ astro tobiasBora ];
     platforms = platforms.all;
```

Comment:
There are many Linux-exclusive dependencies in this package so it won't even evaluate on other platforms (i.e. Darwin). I'll attach a patch for Darwin support soon.

Reply:
Should be correct: [0001-bespokesynth-fix-darwin-support.patch.txt](https://github.com/NixOS/nixpkgs/files/7657121/0001-bespokesynth-fix-darwin-support.patch.txt)

![Screenshot_macOS_2021-12-06_01:02:43](https://user-images.githubusercontent.com/23431373/144770465-f271725e-7a52-40b9-b4d4-36930a5a29ba.png)

Reply:
thanks a lot for this. Unfortunately I'm very busy in the next 2 weeks, would you mind pushing your patch on the branch https://github.com/tobiasBora/nixpkgs/tree/bespoke_synth_zenity as a PR so that I can accept it in a single click? (that way it will be added to this PR) Also, feel free to adapt the `cmakeFlag` as you wish.